docs/migration-guide.md

```diff
@@ -0,0 +1,732 @@
+```
+
+**Progressive Rollout:**
+```yaml
+cutover:
+  strategy: Progressive
+  steps:
+    - rampPercentage: 1
+      pauseDuration: 5m
+    - rampPercentage: 10  
+      pauseDuration: 10m
+    - rampPercentage: 50
+      pauseDuration: 15m
+  gate:
+    workflowType: "HealthCheck"  # Optional validation workflow
+```
+
+### Sunset Configuration
+
+```yaml
+sunset:
+  scaledownDelay: 1h    # Wait 1 hour after draining before scaling to 0
+  deleteDelay: 24h      # Wait 24 hours after draining before deleting
+```
+
+### Resource Management
+
+**CPU and Memory:**
+```yaml
+template:
+  spec:
+    containers:
+    - name: worker
+      resources:
+        requests:
+          memory: "1Gi"
+          cpu: "500m"
+        limits:
+          memory: "2Gi" 
+          cpu: "1"
+```
+
+**Pod Annotations and Labels:**
+```yaml
+template:
+  metadata:
+    labels:
+      team: payments
+      environment: production
+```
+
+## Testing and Validation
+
+### Pre-Migration Testing
+
+1. **Test in non-production environment:**
+   ```bash
+   # Create test TemporalWorkerDeployment
+   kubectl apply -f test-worker.yaml
+   
+   # Verify worker registration
+   kubectl logs -l temporal.io/worker-deployment=test-worker
+   ```
+
+2. **Validate environment variables:**
+   ```bash
+   kubectl exec -it deployment/test-worker-v1 -- env | grep TEMPORAL
+   ```
+
+### Post-Migration Validation
+
+1. **Check version status:**
+   ```bash
+   kubectl get temporalworkerdeployment -o wide
+   ```
+
+2. **Monitor version transitions:**
+   ```bash
+   kubectl get events --field-selector involvedObject.kind=TemporalWorkerDeployment
+   ```
+
+3. **Validate workflow routing:**
+   - Start test workflows
+   - Verify they're routed to correct versions
+   - Check Temporal UI for version distribution
+
+## Common Migration Patterns
+
+Here are common scenarios when migrating from unversioned to versioned workflows:
+
+### Pattern 1: Microservices Architecture
+
+If you have multiple services with their own workers, migrate each service independently:
+
+```yaml
+# Payment service worker
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: payment-processor
+  namespace: payments
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: payments
+  cutover:
+    strategy: Progressive  # Conservative rollout for financial operations
+    steps:
+      - rampPercentage: 5
+        pauseDuration: 10m
+      - rampPercentage: 25
+        pauseDuration: 15m
+  # ... rest of config
+---
+# Notification service worker (separate service, different risk profile)
+apiVersion: temporal.io/v1alpha1
+kind: TemporalWorkerDeployment
+metadata:
+  name: notification-sender
+  namespace: notifications
+spec:
+  workerOptions:
+    connection: production-temporal
+    temporalNamespace: notifications
+  cutover:
+    strategy: AllAtOnce  # Lower risk, faster rollouts acceptable
```

Comment:
maybe "desired" instead of acceptable? IMO if people don't know what they want they should do progressive, so "acceptable" is not a good enough reason to do AllAtOnce